### ISSUE TITLE
clang doesn't warn for missing return value if function contains "if consteval"

### ISSUE THREAD

**Issue by geza-herman** [llvm/llvm-project#116485]
Compile this little snippet, and notice that clang doesn't give a warning for missing `return` (if the `if consteval` is removed, then the warning is properly shown).

```c++
int foo() {
    if consteval {
    }
}
```

version: `Debian clang version 20.0.0 (++20241008102056+4d218caa7716-1~exp1`
